docs/self-managed/operational-guides/production-guide.md

```diff
@@ -0,0 +1,572 @@
+
+```yaml
+identity:
+  serviceAccount:
+    enabled: false
+console:
+  serviceAccount:
+    enabled: false
+webModeler:
+  serviceAccount:
+    enabled: false
+connectors:
+  serviceAccount:
+    enabled: false
+core:
+  serviceAccount:
+    enabled: false
+optimize:
+  serviceAccount:
+    enabled: false
+```
+
+You should only enable the auto-mounting of a service account token when the application explicitly needs access to the Kubernetes API server or you have created a service account with the exact permissions required for the application and bound it to the pod.
+
+- If you have a use case for enabling [Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) then it is recommended to do so.
+<!--Maybe link this to customer: https://github.com/ahmetb/kubernetes-network-policy-recipes-->
+- It is possible to have a pod security standard that is suitable to the security constraints you might have. This is possible through modifying the Pod Security Admission. Please refer to the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-admission/) in order to do so.
+- By default, The Camunda Helm Chart is configured by default to use a read-only root file system for the pod. It is advisable to retain this default setting, and no modifications are required in your `production-values.yaml`.
+- Disable privileged containers. This can be achieved by implementing a pod security policy. For more information please visit the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-admission/)
+- It is possible to modify either the `containerSecurityContext` or the `podSecurityContext`. For example, here is a configuration for the core component that can be added to your `production-values.yaml`:
+
+```yaml
+podSecurityContext:
+  runAsNonRoot: true
+  fsGroup: 1001
+  seccompProfile:
+    type: RuntimeDefault
+
+containerSecurityContext:
+  allowPrivilegeEscalation: false
+  privileged: false
+  readOnlyRootFilesystem: true
+  runAsNonRoot: true
+  runAsUser: 1001
+  seccompProfile:
+    type: RuntimeDefault
+```
+
+- It is recommended to pull images exclusively from a private registry, such as [Amazon ECR](https://aws.amazon.com/ecr/), rather than directly from Docker Hub. Doing so enhances control over the images, avoids rate limits, and improves performance and reliability. Additionally, you can configure your cluster to pull images only from trusted registries. Tools like [Open Policy Agent](https://blog.openpolicyagent.org/securing-the-kubernetes-api-with-open-policy-agent-ce93af0552c3#3c6e) can be used to enforce this restriction.
+- Open Policy Agent can also be used to [whitelist for ingress hostnames](https://www.openpolicyagent.org/docs/latest/kubernetes-tutorial/#4-define-a-policy-and-load-it-into-opa-via-kubernetes).
```

Comment:
maybe do a single point of Open Policy Agent combining this and the previous trusted sources one as sub points?

-  .... recommended to use private registry ...
- ... Open Policy Agent ...
    - trusted registry topic
    - whitelist ingress topic

For the whitelist ingress topic, again why should it be done, what's the advantage as to why we recommend it.

Reply:
> For the whitelist ingress topic, again why should it be done, what's the advantage as to why we recommend it.

It's just a recommendation. I added it because if I was setting up for production, I would want this feature. The reason I recommended it was obviously for enhancing security, but also for error reduction. You reduce the likelihood of misconfigurations related to the hostname.

Reply:
I will create a single point for Open Policy Agent. Thanks.

Reply:
I hope this is clear. Feel free to unresolved and ping me if you would like to discuss this further.